modules/gdscript/gdscript_tokenizer.cpp

```diff
@@ -820,6 +824,19 @@ GDScriptTokenizer::Token GDScriptTokenizer::number() {
 		}
 	}
 
+	if (need_digits) {
+		// No digits in hex or bin literal.
+		Token error = make_error(vformat(R"(Expected %s digit after "0%c".)", (base == 16 ? "hexadecimal" : "binary"), (base == 16 ? 'x' : 'b')));
+		error.start_column = column;
+		error.leftmost_column = column;
+		error.end_column = column + 1;
+		error.rightmost_column = column + 1;
+		push_error(error);
+
+		// Return empty literal.
+		return make_literal((int64_t)0);
```

Comment:
```suggestion
		return error;
```

Reply:
Will do, it prints different errors about expecting an expression but makes sense to error out

Could return errors more here to keep it unified, what do you think?

Reply:
I think `push_error()` is used when we want to continue reading the token despite an error (for example inside string literals) and `return make_error()`/`return error` when the current token has most likely ended.

Reply:
Could swap for the others then too if desired